source/vir/src/modes.rs

```diff
@@ -340,6 +341,25 @@ fn check_expr(typing: &mut Typing, outer_mode: Mode, expr: &Expr) -> Result<Mode
             }
             Ok(mode)
         }
+        ExprX::OpenInvariant(inv, binder, body) => {
+            assert_no_early_exit_in_inv_block(&body.span, body)?;
```

Comment:
This may belong to `well_formed`, unless this check is required for the following mode code.